processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -109,8 +117,18 @@
         private MethodReference factoryMethod;
         private boolean hasFactoryMethod;
 
+        public Builder() {
+            super( Builder.class );
+        }
+
+        @Override
+        protected boolean shouldUsePropertyNamesInHistory() {
+            return true;
+        }
+
         public Builder mappingContext(MappingBuilderContext mappingContext) {
             this.ctx = mappingContext;
+            super.mappingContext( mappingContext );
```

Comment:
Fix all referencing to the ctx to use the super version instead of managing it twice.